examples/deadletterqueue/movingMessagesToDLQ.ts

```diff
@@ -0,0 +1,192 @@
+import {
+  delay,
+  generateUuid,
+  SendableMessageInfo,
+  OnMessage,
+  OnError,
+  MessagingError,
+  ServiceBusMessage,
+  ReceiveMode,
+  Namespace,
+  QueueClient
+} from "../../lib";
+import * as dotenv from "dotenv";
+dotenv.config();
+
+const str = process.env.SERVICEBUS_CONNECTION_STRING || "";
+const queuePath = process.env.QUEUE_NAME || "";
+const numberOfMessages: number = parseInt(process.env.MESSAGE_COUNT || "1");
+const receiveClientTimeout: number = 10000;
+console.log("str: ", str);
+console.log("path: ", queuePath);
+console.log("Number of messages to send each time: ", numberOfMessages);
+
+let ns: Namespace;
+
+/*
+    There are many scenarios wherein messages are moved to the DLQ.
+    Ref: https://docs.microsoft.com/en-us/azure/service-bus-messaging/service-bus-dead-letter-queues
+
+    This sample demonstrates usage of DLQ for 3 such scenarios:
+      A. When client invokes the .deadletter() API on the brokered message.
+      B. When system exceeds maximum allowed delivery attempts on the message,
+         which is by default set to 10
+      C. When processing of message encounters error or an exception.
+
+    After running this sample, run the processMessagesInDLQ example to see how the messages
+    can be reprocessed.
+
+*/
+async function main(): Promise<void> {
+  try {
+    ns = Namespace.createFromConnectionString(str);
+    // Clean up queues before running.
+    await purgeAllQueues();
+
+    // This scenario demonstrates brokered message's deadletter() API resulting in immediate
+    // message move to DLQ.
+    await runSimpleDeadLetterScenario();
+
+    // This scenario demonstrates how exceeding delivery count on message will result in
+    // automatic move to DLQ
+    await runExceedMaxRetriesScenario();
+
+    // This scenario demonstrates how message processing failures can cause message move to DLQ.
+    await runProcessingErrorHandlingScenario();
+  } catch (err) {
+    console.log(">>>>> Error in running sample scenarios: ", err);
+  } finally {
+    console.log("\n >>>> Calling close....");
+    ns.close();
+  }
+}
+
+// Queue clean up util
+async function purgeAllQueues(): Promise<void> {
```

Comment:
In general I think having multiple methods to split up the work is good, however in this case I don't think it is worth having this separate method. We can simply inline the code for it directly under the comment.

That way the sample is simpler and easier to follow.